File: sample-bots/csharp/Walls/Walls.cs
using System;
using System.Drawing;
using Robocode.TankRoyale.BotApi;
using Robocode.TankRoyale.BotApi.Events;

// ------------------------------------------------------------------
// Walls
// ------------------------------------------------------------------
// A sample bot original made for Robocode by Mathew Nelson.
// Ported to Robocode Tank Royale by Flemming N. Larsen.
//
// Moves around the outer edge with the gun facing in.
// ------------------------------------------------------------------
public class Walls : Bot
{
    bool peek; // Don't turn if there's a bot there
    double moveAmount; // How much to move

    // The main method starts our bot
    static void Main()
    {
        new Walls().Start();
    }

    // Constructor, which loads the bot config file
    Walls() : base(BotInfo.FromFile("Walls.json")) { }

    // Called when a new round is started -> initialize and do some movement
    public override void Run()
    {
        // Set colors
        BodyColor = Color.Black;
        TurretColor = Color.Black;
        RadarColor = Color.Orange;
        BulletColor = Color.Cyan;
        ScanColor = Color.Cyan;

        // Initialize moveAmount to the maximum possible for the arena
        moveAmount = Math.Max(ArenaWidth, ArenaHeight);
        // Initialize peek to false
        peek = false;

        // turn to face a wall.
        // `Direction % 90` means the remainder of Direction divided by 90.
        TurnRight(Direction % 90);
        Forward(moveAmount);

        // Turn the gun to turn right 90 degrees.
        peek = true;
        TurnGunRight(90);
        TurnRight(90);

        // Main loop
        while (IsRunning)
        {
            // Peek before we turn when forward() completes.
            peek = true;
            // Move up the wall
            Forward(moveAmount);
            // Don't peek now
            peek = false;
            // Turn to the next wall
            TurnRight(90);
        }
    }

    // We hit another bot -> move away a bit
    public override void OnHitBot(HitBotEvent e)
    {
        // If he's in front of us, set back up a bit.
        var bearing = BearingTo(e.X, e.Y);
        if (bearing > -90 && bearing < 90)
        {
            Back(100);
        }
        else
        { // else he's in back of us, so set ahead a bit.
            Forward(100);
        }
    }

    // We scanned another bot -> fire!
    public override void OnScannedBot(ScannedBotEvent e)
    {
        SetFire(2);
        // Note that scan is called automatically when the bot is turning.
        // By calling it manually here, we make sure we generate another scan event if there's a bot
        // on the next wall, so that we do not start moving up it until it's gone.
        if (peek)
            Rescan();
    }
}

File: sample-bots/python/Walls/Walls.py
from robocode_tank_royale.bot_api.bot import Bot
from robocode_tank_royale.bot_api.color import Color
from robocode_tank_royale.bot_api.events import ScannedBotEvent, HitBotEvent


# ------------------------------------------------------------------
# Walls
# ------------------------------------------------------------------
# A sample bot originally made for Robocode by Mathew Nelson.
#
# This robot navigates around the perimeter of the battlefield with
# the gun pointed inward.
# ------------------------------------------------------------------
class Walls(Bot):
    def __init__(self) -> None:
        super().__init__()
        self._peek: bool = False  # Don't turn if there's a bot there
        self._move_amount: float = 0.0  # How much to move

    def run(self) -> None:
        # Set colors
        self.body_color = Color.BLACK
        self.turret_color = Color.BLACK
        self.radar_color = Color.ORANGE
        self.bullet_color = Color.CYAN
        self.scan_color = Color.CYAN

        # Initialize move_amount to the maximum possible for the arena
        self._move_amount = max(self.arena_width, self.arena_height)
        # Initialize peek to false
        self._peek = False

        # Turn to face a wall
        self.turn_right(self.direction % 90)
        self.forward(self._move_amount)

        # Turn the gun to turn right 90 degrees.
        self._peek = True
        self.turn_gun_left(90)
        self.turn_left(90)

        # Main loop
        while self.running:
            # Peek before we turn when forward() completes.
            self._peek = True
            # Move up the wall
            self.forward(self._move_amount)
            # Don't peek now
            self._peek = False
            # Turn to the next wall
            self.turn_left(90)

    def on_hit_bot(self, e: HitBotEvent) -> None:
        # If he's in front of us, back up a bit; else move ahead a bit.
        bearing = self.bearing_to(float(e.x), float(e.y))
        if -90 < bearing < 90:
            self.back(100)
        else:
            self.forward(100)

    def on_scanned_bot(self, e: ScannedBotEvent) -> None:
        del e
        self.fire(2)
        # Ensure we generate another scan event if we're peeking
        if self._peek:
            self.rescan()


def main() -> None:
    bot = Walls()
    bot.start()


if __name__ == "__main__":
    main()
